components/proxy/src/main/kotlin/com/hotels/styx/routing/db/StyxObjectStore.kt

```diff
@@ -102,8 +102,8 @@ class StyxObjectStore<T> : ObjectStore<T> {
             new = current.minus(key)
         }
 
-        queue {
-            if (current != new) {
+        if (current != new) {
```

Comment:
I'm only seeing the last commit so I haven't reviewed throughly, but I guess you were queueing the comparison to  avoid concurrency issues. Is that not needed?

Reply:
Now I cannot remember but intuitively it feels safe to do this.

Reply:
ah, sure. I hadn't checked the previous code.  Totally fine.